TwoPageEBookReader/app/src/main/java/com/example/ebook_reader_sample/Book.kt

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License.
+ */
+package com.example.ebook_reader_sample
+
+import android.content.Context
+import android.content.res.AssetManager
+import android.graphics.Rect
+import android.text.Html
+import android.view.View
+import android.widget.LinearLayout
+import android.widget.TextView
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import kotlin.math.max
+import kotlin.math.min
+
+class Book(private val filePath: String, private val activityContext: Context) {
+    private val assetManager: AssetManager = activityContext.assets
+    private var chapterStarts: ArrayList<Int> = ArrayList()
+    private var chapterLengths: ArrayList<Int> = ArrayList()
+    val numChapters
+        get() = chapterStarts.size
+    var chapterStartString = activityContext.resources.getString(R.string.chapter_start)
+    var chapterEndString = activityContext.resources.getString(R.string.chapter_end)
+
+    init {
+        val bufferedReader = BufferedReader(InputStreamReader(assetManager.open(filePath)))
```

Comment:
Is better if we  move the whole algorithm (basically the init code) to its own function, so then when we read the code it reads as:
```
init {
    doSomething()
}
```

Reply:
Moved init code into a constructor() function